packages/core/src/components/DataTable/index.tsx

```diff
@@ -122,12 +123,14 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     }
   }
 
-  private getTableHeight = (expandedDataList: ExpandedRow[]) => {
+  private getTableHeight = (expandedDataList: ExpandedRow[]): number => {
     const { height, rowHeight, showAllRows } = this.props;
 
-    return showAllRows
-      ? expandedDataList.length * getHeight(rowHeight) + this.getColumnHeaderHeight()
-      : height || 0;
+    if (showAllRows) {
+      return expandedDataList.length * getHeight(rowHeight) + this.getColumnHeaderHeight();
+    }
+
+    return height!;
```

Comment:
Is this non-null assertion always correct? What if no height prop is provided? `getTableHeight` used to default to 0

Reply:
Yeah, there's a `defaultProp` for `height`, but unfortunately typescript can't pick this up.